tokio/src/io/blocking.rs

```diff
@@ -276,5 +276,24 @@ cfg_fs! {
             self.buf.truncate(0);
             ret
         }
+
+        pub(crate) fn copy_from_bufs(&mut self, bufs: &[io::IoSlice<'_>]) -> usize {
+            assert!(self.is_empty());
+
+            let n = bufs.iter().map(|b| b.len()).sum::<usize>().min(MAX_BUF);
```

Comment:
This iterates all of the buffers every time, even if we only write a few of them. If the buffers are very long and this is called in a loop, that gives quadratic performance.

We should be able to embed this logic inside the for loop instead to avoid that.

Reply:
You mean `b.len()` is `O(n)` and causes the `bufs.iter()` to be `O(n^2)`? I thought since it's a `Deref` to `&[u8]`, it's `O(1)`.
I've provided an alternate implementation that doesn't use `bufs.iter()`.

Reply:
No, `b.len()` is constant time. Instead, it's O(n) in the length of `bufs`, which you iterate over.